martin/src/utils/utilities.rs

```diff
@@ -28,3 +32,38 @@ pub fn encode_brotli(data: &[u8]) -> Result<Vec<u8>, std::io::Error> {
     encoder.write_all(data)?;
     Ok(encoder.into_inner())
 }
+
+pub fn parse_base_path(path: &str) -> MartinResult<String> {
+    if !path.starts_with('/') {
+        return Err(BasePathError(path.to_string()));
+    }
+    if let Ok(uri) = path.parse::<Uri>() {
+        let mut result = uri.path().to_string();
+        while result.chars().last().is_some_and(|v| v == '/') && result.len() > 1 {
```

Comment:
Use pop() back as the trim_end_matches would return "' for "//", any  better way to use the trim methods?

Reply:
ok, lets keep it simple - use this method just for the user-supplied base_path parsing - in which case it should be empty if the user only passed `/`. And keep the original code for x-rewrite-url header.

I will look at it later, see if it can be optimized, but its not worth it now

Reply:
Actually It's not bad to use pop() here?

Reply:
But your suggestion is more simple